okhttp/src/test/java/io/grpc/okhttp/AsyncFrameWriterTest.java

```diff
@@ -0,0 +1,138 @@
+package io.grpc.okhttp;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import io.grpc.internal.SerializingExecutor;
+import io.grpc.okhttp.internal.framed.FrameWriter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executor;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AsyncFrameWriterTest {
+
+  @Mock private OkHttpClientTransport transport;
+  @Mock private Socket socket;
+  @Mock(answer = Answers.CALLS_REAL_METHODS)
+  private PartiallyTrackingFrameWriter frameWriter;
+
+  private QueueingExecutor queueingExecutor;
+  private AsyncFrameWriter asyncFrameWriter;
+
+  @Before
+  public void setUp() throws Exception {
+    queueingExecutor = new QueueingExecutor();
+    asyncFrameWriter = spy(new AsyncFrameWriter(transport, new SerializingExecutor(queueingExecutor)));
```

Comment:
Please, please no spies. They are so very magical, I don't trust tests with them.

Reply:
spy is gone!